XLS-0045-prepublish-validator-lists/README.md

```diff
@@ -0,0 +1,824 @@
+`blob` will be considered malformed. Clients MAY process the other
+`blob`s in the file as if the blob with that sequence were not present.
+
+If the `manifest` is not present in a `blobs-v2` array entry, then the
+top-level `manifest` will be used when checking the signature. This allows
+a publisher to keep any old, `blob`s and their `signature`s if they ever need
+to update their manifest. Most publishers will only need to use this field
+very rarely. (Currently, if the VL processing sees a `stale` manifest, where
+the public key hasn't been completely revoked, it will be used without a
+problem. I'm not sure this behavior is correct. Consider treating a
+`stale` manifest as an `untrusted` list.)
+
+Pros of the new format:
+* By signing each `blob` separately, the VL file can be modified to
+  move the `blob`s around in the `blobs-v2` array as older `blob`s expire
+  and new `blob`s are added without requiring any extra signing.
+* The objects in the `blobs-v2` array don't necessarily need to be in
+  order, but it will simplify maintenance if they are.
+* Old `blob`s can be left in the `blobs-v2` array without causing any harm, as
+  long as a more recent blob has already been added. In other words, old data
+  can be cleaned up when it's convenient.
+
+Cons:
+* Each `blob` will need to be signed separately, even if they are all built at
+  once.
+
+##### Migration strategy
+
+**Don't implement any migration strategy in code**
+
+This is the simplest option. Don't do anything with the files, don't
+modify the URLs. Leave the decision of when and how to migrate to
+publishers. If they want to publish both versions simultaneously, then
+they will need to communicate different URLs to operators. If not, then
+they can wait until an appropriate time (e.g. when enough nodes have
+upgraded, or if they want to pressure nodes to upgrade), and then make
+a single clean switchover from v1 to v2.
+
+In terms of the code, if the file version is 1, then rippled will follow
+the v1 rules, and if it's 2, rippled will follow the v2 rules.
+
+Pros:
+* Keeps rippled code simple
+* If a publisher chooses to publish a v2 UNL file while there are still
+  nodes on the network that won't understand it, newer peers will
+  [communicate](#protocol-changes-for-v2-unl-broadcasts) the "current"
+  UNL to older peers, if they're connected sufficiently.
+
+Cons:
+* Adoption will probably be delayed until some other "limiting event" or
+  synchronization point happens which forces all nodes to upgrade. This
+  will most likely be when an amendment is enabled that no v1-only nodes
+  support.
+
+See the [appendix](#appendix) for descriptions of some other migration
+strategies that were considered and rejected.
+
+-------------------------------------------- ---------
+:bangbang: Since this code was merged, there have been several amendments enabled on Mainnet,
+  so there are no versions of rippled that don't support v2.
```

Comment:
```suggestion
:bangbang: Support for XLS-45 has been fully deployed across all XRPL Mainnet
nodes.

This is because, after the "UNLv2" implementation was included in a release,
amendments from that release (or later) have been enabled. For details, see
[Amendment Blocked Servers](https://xrpl.org/docs/concepts/networks-and-servers/amendments/#amendment-blocked-servers).
```